paasta_tools/cli/cmds/get_deploygroup_history.py

```diff
@@ -0,0 +1,177 @@
+        help="define a different soa config directory",
+    )
+
+    list_parser.set_defaults(command=paasta_get_deploygroup_history)
+
+
+def get_versions_for_service(
+    service: str, deploy_groups: Collection[str], soa_dir: str
+) -> Mapping[DeploymentVersion, Tuple[str, str]]:
+    """Returns a dictionary of 2-tuples of the form (timestamp, deploy_group) for each version tuple of (deploy sha, image_version)"""
+    if service is None:
+        return {}
+    git_url = get_git_url(service=service, soa_dir=soa_dir)
+    all_deploy_groups = list_deploy_groups(service=service, soa_dir=soa_dir)
+    deploy_groups, _ = validate_given_deploy_groups(all_deploy_groups, deploy_groups)
+    previously_deployed_versions: Dict[DeploymentVersion, Tuple[str, str]] = {}
+
+    for ref, sha in list_remote_refs(git_url).items():
+        regex_match = extract_tags(ref)
+        try:
+            deploy_group = regex_match["deploy_group"]
+            tstamp = regex_match["tstamp"]
+            image_version = regex_match["image_version"]
+        except KeyError:
+            pass
+        else:
+            if deploy_group in deploy_groups:
+                version = DeploymentVersion(sha=sha, image_version=image_version)
+                previously_deployed_versions[version] = (tstamp, deploy_group)
+    return previously_deployed_versions
+
+
+def list_previous_versions(
+    service: str,
+    deploy_groups: Collection[str],
+    any_given_deploy_groups: bool,
+    versions: Mapping[DeploymentVersion, Tuple],
+) -> None:
+    def format_timestamp(tstamp: str) -> str:
+        return naturaltime(datetime_from_utc_to_local(parse_timestamp(tstamp)))
+
+    print("Below is a list of deploy_groups History:")
+    # All versions sorted by deployment time
+    list_of_versions = sorted(versions.items(), key=lambda x: x[1], reverse=True)
+    rows = [("Timestamp -- UTC", "Human time", "deploy_group", "Version")]
+    for version, (timestamp, deploy_group) in list_of_versions:
+        rows.extend(
+            [(timestamp, format_timestamp(timestamp), deploy_group, repr(version))]
+        )
+    for line in format_table(rows):
+        print(line)
+
+
+def paasta_get_deploygroup_history(args: argparse.Namespace) -> int:
+    soa_dir = args.soa_dir
+    service = figure_out_service_name(args, soa_dir)
+
+    deploy_info = get_deploy_info(service=service, soa_dir=args.soa_dir)
+    if not can_user_deploy_service(deploy_info, service):
+        return 1
+
+    if args.all_deploy_groups:
+        given_deploy_groups = list_deploy_groups(service=service, soa_dir=soa_dir)
+    else:
+        given_deploy_groups = {
+            deploy_group
+            for deploy_group in args.deploy_groups.split(",")
+            if deploy_group
+        }
+
+    all_deploy_groups = list_deploy_groups(service=service, soa_dir=soa_dir)
+    deploy_groups, invalid = validate_given_deploy_groups(
+        all_deploy_groups, given_deploy_groups
+    )
+
+    if len(invalid) > 0:
```

Comment:
up to you: but this could also be written as `if invalid_deploy_groups` since an empty collection is falsey while a non-empty one is truthy

Reply:
(if you end up liking this, you'd probably also want to update `if len(deploy_groups) == 0` below to `if not deploy_groups`)